### DISCUSSION TITLE
Cannot Get API Webpage to Appear

### DISCUSSION THREAD

**Discussion by zveltri** [cmu-sei/GHOSTS#97] [General]
Hello,

I have followed the directions and everything seems to be set up correct other than I am unable to access the API webpage. I attempt to visit HTTP://localhost:5000/api/home and the following message appears: "Unable to Connect. This site could be temporarily unavailable or too busy. If you still cannot load the webpage, check the machine's internet connection." 

I know the machine has internet. It is the server I have setup Grafana and everything else on.

Also, once I load some test clients into the environment, where or what file are the logs stored? I can see everything within the Grafana dashboard, but haven't been able to find any logs.

**Comment by Remraf1** [comment]
Not sure if it helps, but to view my api/home I had to use port :52388 in my browser.  5000 didn't work for me either, but I have a client up and running.

**Reply by zveltri** [reply]
That worked perfectly. I can now see the API page. Would you happen to know why my client isn't reporting any data to grafana? I am able to access the grafana page via port 3000

**Reply by Remraf1** [reply]
Have you added the postgres datasource to grafana yet?  I had to do it manually, but I'm also really unfamiliar with grafana lol.  I used the info in the datasources.yml file to add it as a data source, then I imported the Ghost5 dashboard.

https://github.com/cmu-sei/GHOSTS/tree/master/configuration/grafana

**Reply by zveltri** [reply]
Yes, I added the datasource with its IP address followed by :5432. I also imported the dashboard json

**Reply by zveltri** [reply]
I finally have the client reporting to Grafana. NPCs 1 and Polling 2....The graphs are not populating still, or does this take awhile? I am not familiar with this software either.

**Reply by sei-dupdyke** [reply]
Hey sorry, taking a few days PTO and just saw this.

If you are running via docker, do a `docker ps --all` in the terminal. Is Postgres and the Ghosts API running? Can you hit it on http://localhost:5000 on the server machine?

Server logs are viewed via `docker logs -f ghosts-api` (assuming docker and you want to see the API logs). Client logs are stored in ./logs until they are beamed up to the server periodically.

**Reply by zveltri** [reply]
Yes, I am running via docker. Also, yes I can see the API server but on port :52388.  I'll take a look at those locations after I let these test machines run some. Thank you.

**Comment by sei-dupdyke** [comment]
Hmm, if you point to http://localhost:52388/api/home, you're probably good to go.

If you want to change that to :5000 or any other port, do this:

1. in the terminal `docker-compose down`
2. now edit the docker-compose.yml file -> this line: https://github.com/cmu-sei/GHOSTS/blob/01aa57495849d725005a43b9b7ff0155aca6e14e/src/Ghosts.Api/docker-compose.yml#L41 and change `52388` to your preferred port.
3. `docker-compose up -d` and you're on your port now

Sorry about that, my personal settings somehow made it into the release. Typically the port is 5000, not 52388.
